### PR TITLE
feat: Add test scenarios for North-South Messaging

### PR DESCRIPTION
**Pull request by cherrycl**
Scenarios only.
#685 
Signed-off-by: Cherry Wang <cherry@iotechsys.com>

**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [ ] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?)
- [ ] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?)
  <link to docs PR>

### CODE REVIEW

**Review by jumpingliu — DISMISSED** [review]
LGTM

File: TAF/testScenarios/integrationTest/UC_north_south_messaging/GET-Negative.robot [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+*** Test Cases ***
+ErrNSMessagingGET001 - Query Commands with non-existent device name
+    Given Subscribe MQTT Topic 'edgex/commandquery/response'
+    When Get Command With non-existent device From External MQTT Broker
+    Then Should Return Error Code "1"
+    And RequestID Should Be Correct
+
+ErrNSMessagingGET002 - Get specified device non-existent read command
+    Given Create Device For device-virtual
+    And Subscribe MQTT Topic 'edgex/commandquery/response'
```
Comment (hahattan) [inline TAF/testScenarios/integrationTest/UC_north_south_messaging/GET-Negative.robot:21]: Subscribe to `edgex/command/response/#` for command response,  same for below cases

File: TAF/testScenarios/integrationTest/UC_north_south_messaging/GET-Negative.robot [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    When Get Command With non-existent command From External MQTT Broker
+    Then Should Return Error Code "1"
+    And RequestID Should Be Correct
+    [Teardown]  Delete device by name
+
```
Comment (hahattan) [inline TAF/testScenarios/integrationTest/UC_north_south_messaging/GET-Negative.robot:26]: Add a scenario of querying all device commands with invalid offset or limit parameter

**Review by hahattan — CHANGES_REQUESTED** [review]

File: TAF/testScenarios/integrationTest/UC_north_south_messaging/GET-Negative.robot (hunk 1 above)
Comment (cherrycl) [reply TAF/testScenarios/integrationTest/UC_north_south_messaging/GET-Negative.robot]: Updated.

File: TAF/testScenarios/integrationTest/UC_north_south_messaging/GET-Negative.robot (hunk 2 above)
Comment (cherrycl) [reply TAF/testScenarios/integrationTest/UC_north_south_messaging/GET-Negative.robot:32]: Updated.

File: TAF/testScenarios/integrationTest/UC_north_south_messaging/GET-Negative.robot [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+*** Test Cases ***
+ErrNSMessagingGET001 - Query all DeviceCoreCommands with invalid offset
+    Given Subscribe MQTT Topic 'edgex/commandquery/response'
+    When Query Commands With invalid offset For All Devices From External MQTT Broker
+    Then Should Return Error Code "1"
+    And RequestID Should Be Correct
+
+ErrNSMessagingGET002 - Query Commands with non-existent device name
+    Given Subscribe MQTT Topic 'edgex/command/response/#'
+    When Get Command With non-existent device From External MQTT Broker
```
Comment (hahattan) [inline TAF/testScenarios/integrationTest/UC_north_south_messaging/GET-Negative.robot:21]: I'm confused whether this is a query command or a device command ...
so I think before the AdminState lock case we need:
- Query all with invalid limit/offset (case1)
- Query by device name with non-exist device name (case2 ...?)
- Get device command with non-exist device name (case3)
- Get device command with non-exist command name

Comment (cherrycl) [reply TAF/testScenarios/integrationTest/UC_north_south_messaging/GET-Negative.robot]: It should be query command. Updated the error and added more cases for the function.

**Review by hahattan — APPROVED** [review]
LGTM

**Review by cloudxxx8 — APPROVED** [review]
LGTM

**Pull request merged**